Restore init for clocked discretized. (#3577)
modelica/ModelicaSpecification: chapters/synchronous.tex

\section{Initialization of Clocked Partitions}\label{initialization-of-clocked-partitions}

> The standard scheme for initialization of Modelica models does not apply for discrete-time sub-partitions.

Reviewer: Didn't we drop the _sub_ here?  I can't find a single example of _discrete-time sub-partition_ in the current text?
```suggestion
The standard scheme for initialization of Modelica models does not apply for discrete-time partitions.
```

Instead, initialization is performed in the following way:
\begin{itemize}
\item
  Variables in discrete-time sub-partitions cannot be used in initial equation or initial algorithm sections.
\item

>   Attribute \lstinline!fixed! cannot be applied to variables in discrete-time sub-partitions.

Reviewer: Should we really allow `fixed` to be applied to clocked states in clocked discretized continuous-time partitions?  Wouldn't that conflict with `fixed` already being defined for these variables?

Author: It shouldn't be a conflict! Instead they should keep their "earlier" fixed-ness. (Technically we cannot know whether the fixed-attribute was set before or after it was discretized.) Exactly how that works isn't clear in general.

>   The attribute \lstinline!fixed! is true for variables to which \lstinline!previous! is applied, otherwise false.

Reviewer: While we are at it, shouldn't we avoid the problem when `previous` is applied to a parameter or constant?
```suggestion
  The attribute \lstinline!fixed! is true for clocked states, otherwise false.
```
